global-script.lua

```diff
@@ -304,6 +324,13 @@ function onLoad(saved_data)
     if saved_data ~= "" then
         local loaded_data = JSON.decode(saved_data)
         gameStarted = loaded_data.gameStarted
+        if loaded_data.playerBlocks then
+            playerBlocks = loaded_data.playerBlocks
```

Comment:
`playerBlocks` isn't used after the game starts right?

Reply:
Correct, but save data may exist from before a game starts -- someone may choose the wrong spirit, then undo for instance.  In a started game this is just an empty table anyways so no real space requirements.

Reply:
so when the mod goes back in time, what are you saying happens to the save data?

Reply:
Every undo point is a save.  Undoing is exactly the same as having saved X seconds in the past and reloading it.

While it's unlikely someone's going to create a hard save of a game that hasn't actually started yet, we shouldn't discount it and the fact that someone might undo something like a spirit selection or whatnot before the game starts means it's important to ensure that the color swapping code does not break when loading a save.

Reply:
so i'm pretty sure that basically the entire setup UI will break with undoing. do people expect rewind to work? for as long as i can remember thought it was accepted that you can't rewind any scripted mod. while not the greatest mentality to have, do people even actually try to rewind things?

Reply:
The setup UI doesn't break at all on an undo, it just 'resets' to its initial state (requiring you to reselect all options.)

In my case, things _would_ break unless either *all* of it saves or *none* of it does.  Since some positions are swapped (and that's saved automagically), it's necessary to persist the rest.  Plus, I feel Undo could be more useful and wouldn't want to actively work against it.